sdk/identity/azure-identity/src/main/java/com/azure/identity/AzurePipelinesCredentialBuilder.java

```diff
@@ -108,6 +109,19 @@ public AzurePipelinesCredential build() {
 
         String requestUrl = String.format("%s?api-version=%s&serviceConnectionId=%s",
             oidcEndpoint, OIDC_API_VERSION, serviceConnectionId);
+
+        // We need to make a best effort to log these headers.
+        // If the user gives us options, we'll sneak the headers in, or make an options with them.
+        // If the user gave us a pipeline none of this matters as we can't change a user created pipeline.
+        if (identityClientOptions.getHttpPipeline() != null) {
+            HttpLogOptions options = identityClientOptions.getHttpLogOptions();
+            if (options == null) {
+                options = new HttpLogOptions();
+            }
+            options.addAllowedHeaderName("x-vss-e2eid");
+            options.addAllowedHeaderName("x-msedge-ref");
+            identityClientOptions.setHttpLogOptions(options);
```

Comment:
Reminder to myself, add overloads in `HttpLogOptions` that use `HttpHeaderName` when it comes to header logging.

Reply:
#42073 :)